### DISCUSSION TITLE
Warning: Prop `className` did not match. Server: “x” Client: “y” with styled components

### DISCUSSION THREAD

**Discussion by paulius1990** [vercel/next.js#17866] [Help]
In app when i change something in the code, the HMR works and shows the correct update but if I refresh the page, this error comes up again. This happens on dev mode. Noticed where are a lot of topics with this error, tried all day different configuration setups with no effort. Clearing my cache/restarting my dev server seems to have fix the issue on initial load, but not  Maybe it's to due with Next.js.

Please help me to get rid of the error.

> Warning: Prop className did not match. Server: "sc-cBoprd hjrjKw" Client: "sc-iCoHVE daxLeG"

_App.tsx
```
function MyApp({ Component, pageProps, mainController }) {
  return (
    <ConfigurationProvider configuration={configuration}>
        <ThemeProvider theme={mainController.getTheme()}>
          <Normalize />
          <Font />
          <Component {...pageProps} controller={mainController} />
        </ThemeProvider>
    </ConfigurationProvider>
  );
}

export default appControllerContext(MyApp);
```

_document.tsx
```
import Document from 'next/document'
import { ServerStyleSheet } from 'styled-components'

export default class MyDocument extends Document {
  static async getInitialProps(ctx) {
    const sheet = new ServerStyleSheet()
    const originalRenderPage = ctx.renderPage

    try {
      ctx.renderPage = () =>
        originalRenderPage({
          enhanceApp: (App) => (props) =>
            sheet.collectStyles(<App {...props} />),
        })

      const initialProps = await Document.getInitialProps(ctx)
      return {
        ...initialProps,
        styles: (
          <>
            {initialProps.styles}
            {sheet.getStyleElement()}
          </>
        ),
      }
    } finally {
      sheet.seal()
    }
  }
}
```

.babelrc
```
{
  "presets": [
    "next/babel"
  ],
  "plugins": [
    [
      "babel-plugin-styled-components",
      {
        "ssr": true,
        "displayName": true,
        "preprocess": false
      }
    ]
  ]
}
```

**_Versions_**:

```
"next": "9.5.5",
"styled-components": "5.2.0",
"babel-plugin-styled-components": "1.11.1",
```

**Comment by jamesmosier** [comment]
Your setup looks fine, so the issue will be coming from somewhere else in your codebase.

Try searching in your browsers devtools for `sc-iCoHVE daxLeG` as a class name and then that will help to inform you of the component that is causing this issue. It could be any number of things but without seeing the code it is hard to say.
